### ISSUE TITLE
Unexpected behavior when using np.int64 instead of int when creating a graph

### ISSUE THREAD

**Issue by leotrs** [networkx/networkx#5582]
When trying to create a lollipop graph, I get different results depending on whether the arguments passed are `int` or `np.int64`.

### Current Behavior
Two DIFFERENT graphs are returned, even when the arguments are numerically the same.

### Expected Behavior
The two graphs should be the same.

### Steps to Reproduce
```python
>>> G = nx.lollipop_graph(3, 3)
>>> H = nx.lollipop_graph(np.int64(3), np.int64(3))
>>> G.order(), G.size() # correct, expected
(6, 6)
>>> H.order(), H.size() # wrong, unexpected
(3, 3)
```

### Environment

Python version: 3.8.10
NetworkX version: 2.5
numpy version: 1.20.2

### Additional context
This happened when trying to create multiple random lollipop graphs like thus:
```python
for n1, n2 in np.random.randint(3, 20, size=(100, 2)):
    G = nx.lollipop_graph(n1, n2)
```
`np.random.ranint` usually returns an `int`, but when used to generate an array instead of a single number, the type of each number is `np.int64`, as the following code shows:
```python
>>> for n1, n2 in np.random.randint(3, 20, size=(100, 2)):
...     print(type(n1)) 
<class 'numpy.int64'>
```

**Comment by dschult** [comment]
I can verify this defect.  Thanks very much.
It looks like checking `isinstance(m, int)` doesn't work for `np.int64`.
We may want to add a conversion to `int` in the `nodes_or_number` decorator that processes `m`.
But is there a standard way to check numpy values for whether they are integers or not?

**Comment by rossbar** [comment]
>  It looks like checking isinstance(m, int) doesn't work for np.int64.

Yes, that would be expected - `np.int64` is a fixed-width integer type whereas the Python int is not. A better way to check if something is *any* integer type is to use the `Integral` abc, e.g.

```python
>>> from numbers import Integral
>>> isinstance(value, Integral)
```

**Comment by dschult** [comment]
Thanks @rossbar! 
I think in this case, the `isinstance` is being used to determine whether the input successfully executed `list(range(n))` used inside the `nodes_or_number` decorator. 
Time for an audit of the use of `nodes_or_number`   :}

Potential problems:
- The functions that actually use the 1st value returned by `nodes_or_number` are:
   - `star_graph`, `wheel_graph`, `lollipop_graph`, `complete_bipartite_graph`
- `wheel_graph("")` and `star_graph("")` raise an `IndexError`. Same for any empty sequence input.
-  `lollipop_graph` raises if (m, n) are type (int, seq) e.g. `lollipop_graph(3, "abc")`
- `complete_bipartite_graph` raises if (n1, n2) are type (seq, int) e.g. `complete_bipartite_graph("abc", 3)`

Potential fixes:
- `star_graph` and `lollipop_graph` and `complete_bipartite_graph` use the original input value to check if it was used to construct `list(range(n))`.  These should use `isinstance(n, numbers.Integral)` and the last one already does.
- `wheel_graph` should use `if len(nodes) == 0:` instead of `if n_name == 0:`
- The following shouldn't use `nodes_or_number` because they just call `empty_graph` anyway:  
   - `complete_graph`
   - `cycle_graph`
- unused first values of `nodes_or_number` should unpack it as `_, nodes = n` using `_` as variable name.
- `lollipop_graph` should rename `n_nodes` when `n` is Integral, not `m` as it currently does. 
- `complete_bipartite_graph` should check both n1 and n2 for Integral, should use `len(top)` instead of n1 and should maybe raise a ValueError if top and bottom are not distinct.
- Add tests for the last 3 potential problems listed above. Add tests for non-int Integral inputs for `star_graph`, `lollipop_graph` and `complete_bipartite_graph`.
- Adjust doc_string of `complete_bipartite_graph` to explain the case when n1, n2 are not integers.
